### DISCUSSION TITLE
Just a question, in class `AsyncSession`, why is `add` method `def` but `delete` method `async def`?

### DISCUSSION THREAD

**Discussion by monchin** [sqlalchemy/sqlalchemy#11618] [Usage Questions] [answered]
Hi, I'm now using `2.0.31`, and I found [`AsyncSession.add` ](https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_31/lib/sqlalchemy/ext/asyncio/session.py#L1120) is `def` but [`AsyncSession.delete`](https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_31/lib/sqlalchemy/ext/asyncio/session.py#L758) is `async def`. Just curious about why it is designed like this. Intuitively I suppose `add` and `delete` should be the same, both `def` or both `async def`.

**Comment by monchin** [comment]
Sorry, I saw the comment that `delete may need to cascade along unloaded relationships, it is awaitable to allow for those queries to take place.` So the key is `add` is anyway very fast, but `delete` maybe very slow. Am I right?

**Reply by CaselIT** [reply, marked as answer]
If by very slow you mean "may use the db" the answer is yes.
You can control the relationship behaviour using https://docs.sqlalchemy.org/en/20/orm/cascades.html#passive-deletes

**Reply by monchin** [reply]
Thank you for your kind reply! :)
